tests/monitoring_client.c

```diff
@@ -38,8 +38,10 @@ static struct json_object *json_send_and_receive(int sockfd, int request)
 	json_object_object_add(json_req, "request", json_object_new_int(request));
 
 	const char *req = json_object_to_json_string(json_req);
-
-	ret = send(sockfd, req, strlen(req), 0);
+	char buf[1024];
+	strcpy(buf, req);
+	buf[strlen(req)] = '\0';
+	ret = send(sockfd, req, strlen(buf), 0);
```

Comment:
I don't think we need zero byte in the socket. You still rely on `strlen(req)` and it doesn't change anything.